### PR TITLE
🪂 Continuous Deployment

### PR DESCRIPTION
**Pull request by mrharpo**
# Continuous Deployment
Adds continuous deployment with `build.yml` action from [clamsproject/.github](https://github.com/clamsproject/.github)

## Steps
- Build docker image
- Extract tags and labels
- Login to ghcr.io
- Push image

## On:
- push to `master`
- pull_request to `master`
- release:
  - published
  - edited
  - prereleased

# ghcr.io
Publishes image as `ghcr.io/clamsproject/app-barsdetection`

https://github.com/clamsproject/app-barsdetection/pkgs/container/app-barsdetection

### CODE REVIEW

**Comment by keighrim** [comment]
I'd like to add more actions to the dotgithub repo for and may want to change the name of `build.yml` for distinction.

File: .github/workflows/CD.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: 🪂 Continuous Deployment
+
+on:
+  push:
```
Comment (keighrim) [inline .github/workflows/CD.yml:4]: I'm not sure all these triggers are necessary. We've had an undocumented convention to use git tags for new version releases of apps, so I guess we can trigger this only upon pushing new tags.

Comment (mrharpo) [reply .github/workflows/CD.yml:4]: From a _"Continuous"_ deployment perspective, I've found it is handy to have the images built and pushed more often.

## `on: push: master`
So we can easily deploy bugfixes without versioning a new tag number.

(Side note, we should change `master` to `main` here and in all `apps-*`)

## `pull_request: master`
This is helpful for testing the image of a new feature, before pulling it in.

## `releases`
If clamsproject is not using releases, we can remove this, but we should probably issue regular releases for major stable versions.

## `tags`
We can add triggers on tags, e.g:
```
    tags:
      - '*'
```
... to build on any tag push.

I'd still like to keep the other triggers in for now. If we need to prune old images in the future, we can create a workflow Action for that as well.

**Comment by keighrim** [comment]
Superseded by #3.

**Pull request closed without merging**